sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -615,14 +615,23 @@ sycl::detail::pi::PiProgram ProgramManager::getBuiltPIProgram(
   // Check that device supports all aspects used by the kernel
   if (auto exception = checkDevSupportDeviceRequirements(Device, Img))
     throw *exception;
-
-  auto BuildF = [this, &Img, &Context, &ContextImpl, &Device, Prg, &CompileOpts,
-                 &LinkOpts, SpecConsts] {
+  std::string CompileOptsString, LinkOptsString;
+  if (!CompileOpts.empty())
+    CompileOptsString = std::string(CompileOpts);
+  if (!LinkOpts.empty())
+    LinkOptsString = std::string(LinkOpts);
```

Comment:
The problem is, the string_view we get from applyCompileOptionsFromEnvironment() points to a const char* which means it can't be updated. In BuildF() both CompileOpts and LinkOpts can be modified in applyOptionsFromImage()- which does a bunch of string modifications. Hence, passing a string instead of a string_view to this function is better. After these options are updated we create a new string_view that points to the updated options. 

Are you suggesting that we pass string_view to BuildF() and create strings inside the function before calling applyOptionsFromImage()?

Reply:
> Are you suggesting that we pass string_view to BuildF() and create strings inside the function before calling applyOptionsFromImage()?

Yes. Otherwise, we make a copy regardless of whether BuildF is called or not.

Reply:
Moved the string creation inside BuildF.